internal/cmd/build/main.go

```diff
@@ -0,0 +1,314 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"io/fs"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"runtime"
+	"sort"
+	"strings"
+
+	_ "github.com/BurntSushi/toml"
+	_ "github.com/kisielk/errcheck/errcheck"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/testsuite"
+	_ "honnef.co/go/tools/staticcheck"
+)
+
+func main() {
+	if err := newBuilder().run(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+const coverageDir = ".build/coverage"
+
+type builder struct {
+	thisDir string
+	rootDir string
+}
+
+func newBuilder() *builder {
+	var b builder
+	// Find the root directory from this directory
+	_, thisFile, _, _ := runtime.Caller(0)
+	b.thisDir = filepath.Join(thisFile, "..")
+	b.rootDir = filepath.Join(b.thisDir, "../../../")
```

Comment:
You could call into `git rev-parse --show-toplevel` instead. We know it's a git repo

Reply:
I think this code is much clearer than making a git call. It's really easy to just get a relative directory of a current file.